stl/inc/xatomic_wait.h

```diff
@@ -0,0 +1,128 @@
+// xatomic.h internal header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _XATOMIC_WAIT_H
+#define _XATOMIC_WAIT_H
+#include <yvals.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#include <xatomic.h>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+
+enum _Atomic_spin_phase : unsigned long {
+    _Atomic_wait_phase_mask            = 0x0000'00FF,
+    _Atomic_spin_value_mask            = ~_Atomic_wait_phase_mask,
+    _Atomic_spin_value_step            = _Atomic_wait_phase_mask + 1,
+    _Atomic_wait_phase_init_spin_count = 0x0000'0000,
+    _Atomic_wait_phase_spin            = 0x0000'0008,
+    _Atomic_wait_phase_wait_locked     = 0x0000'0001,
+    _Atomic_wait_phase_wait_none       = 0x0000'0002,
+    _Atomic_wait_phase_wait_counter    = 0x0000'0004,
+    _Atomic_wait_phase_yield           = 0x0000'0010,
+    _Atomic_wait_phase_sleep           = 0x0000'0020,
+    _Atomic_unwait_needed              = _Atomic_wait_phase_wait_locked,
+};
+
+struct _Atomic_wait_context_t {
+    static constexpr unsigned long long _No_deadline = 0xFFFF'FFFF'FFFF'FFFF;
+    unsigned long _Wait_phase_and_spin_count         = _Atomic_wait_phase_init_spin_count;
+    unsigned long long _Deadline                     = _No_deadline; // or GetTickCount64 plus duration

```

Comment:
Add nano resolution, could be used in future Windows or in future CPUs wiht `tpause` instruction.